.github/workflows/upload-and-deploy-to-prod-main.yaml

```diff
@@ -0,0 +1,82 @@
+name: upload-and-deploy-to-prod-main
+
+description: >-
+  Upload artifacts and deploy stacks to main when a new tag is pushed.
+  Tags must be formatted using semantic versioning.
+
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+
+env:
+  NAMESPACE: main
+  PYTHON_VERSION: 3.9
+
+jobs:
+
+  pre-commit:
+    name: Run pre-commit hooks against all files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+      - uses: pre-commit/action@v3.0.0
+
+
+  upload-files:
+    name: Upload files to S3 bucket in prod/main
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    environment: prod
+    steps:
+
+      - name: Setup code, pipenv, aws
+        uses: Sage-Bionetworks/action-pipenv-aws-setup@v3
+        with:
+          role_to_assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          role_session_name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          python_version: ${{ env.PYTHON_VERSION }}
+
+      - name: Copy files to templates bucket
+        run: python src/scripts/manage_artifacts/artifacts.py --upload --namespace $NAMESPACE --cfn_bucket ${{ vars.CFN_BUCKET }}
+
+
+  sceptre-deploy-main:
+    name: Deploys to prod/main using sceptre
+    runs-on: ubuntu-latest
+    needs: [pre-commit, upload-files]
+    environment: prod
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Setup code, pipenv, aws
+        uses: Sage-Bionetworks/action-pipenv-aws-setup@v3
+        with:
+          role_to_assume: ${{ vars.AWS_CREDENTIALS_IAM_ROLE }}
+          role_session_name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          python_version: ${{ env.PYTHON_VERSION }}
+
+      - name: Create directory for remote sceptre templates
+        run: mkdir -p templates/remote/
+
+      - name: Deploy sceptre stacks to prod/main
+        run: pipenv run sceptre --var "namespace=${{ env.NAMESPACE }}" launch prod --yes
+
+      - name: Configure S3 to Glue lambda with S3 trigger
```

Comment:
Was there a reason to leave this lambda trigger action out of the `upload-and-deploy` for dev?

Reply:
I wasn't sure if it would be useful. It could be a nuisance if every time we move data into a namespace in the input bucket the lambda is triggered. Should I be doing this in dev, too?

Reply:
True, I think the main reason I was thinking of having it in dev as well to mirror the structure is if the `lambda action` or something else related to the lambda trigger broke when we are deploying something in dev, we could catch it then (plus one less difference to keep track of between dev and prod). I'm not sure if that's enough of a reason to add it to dev?

Reply:
Yep, I added it to dev.